test/zdtm/static/pidfd00.c

```diff
@@ -0,0 +1,64 @@
+#include <errno.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <poll.h>
+
+#include "zdtmtst.h"
+
+static int pidfd_open(pid_t pid, unsigned int flags)
+{
+	return syscall(SYS_pidfd_open, pid, flags);
+}
+
+const char *test_doc = "Pidfd ";
+const char *test_author = "Suraj Shirvankar <surajshirvankar@gmail.com>";
+
+int main(int argc, char **argv)
+{
+	int pidfd, errcode = 42;
+	int status;
+	int ret;
+	struct pollfd pollfd;
+
+	test_init(argc, argv);
+
+	pidfd = pidfd_open(1, 0);
```

Comment:
The test has to create more than one pidfd linked to different tasks and then checks that they are linked to proper tasks after C/R.

Reply:
@avagin I tried to write a test that would create a fork of a process and the parent process would monitor the pid of the child process. The dumping and restore process would first restore the parent process and the child process would still not be restored, hence the restore would fail.
I was thinking of restoring the pidfd after all the child processes have been restored, does that make sense.
To do this I would have to dump the pidfd seperate from the regular fd dump process.